### DISCUSSION TITLE
Text Detection Dataset Annotation & end-to-end evaluation

### DISCUSSION THREAD

**Discussion by Jeep-Z** [PaddlePaddle/PaddleOCR#15400] [Q&A] [answered]
- I wanna finetune text detection model and use 4 points polygon annotation. Should it be converted to rotated rectangle? Or 4 points polygon annotation also supported?
- Is there an end-to-end evaluation module?

**Comment by liuhongen1234567** [comment, marked as answer]
Hello, the default annotation for the text detection model is a 4-point box, and there is no need for a rotated detection box. For detailed training, please refer to section 4.1 “Dataset and Pretrained Model Preparation.” in https://paddlepaddle.github.io/PaddleOCR/latest/en/version3.x/module_usage/text_detection.html#4-custom-development
The specific label format is as follows:
```python
images/train_img_61.jpg	[{"transcription": "###", "points": [[427, 293], [469, 293], [468, 315], [425, 314]]}, {"transcription": "###", "points": [[480, 291], [651, 289], [650, 311], [479, 313]]}, {"transcription": "Ave", "points": [[655, 287], [698, 287], [696, 309], [652, 309]]}, {"transcription": "West", "points": [[701, 285], [759, 285], [759, 308], [701, 308]]}, {"transcription": "YOU", "points": [[1044, 531], [1074, 536], [1076, 585], [1046, 579]]}, {"transcription": "CAN", "points": [[1077, 535], [1114, 539], [1117, 595], [1079, 585]]}, {"transcription": "PAY", "points": [[1119, 539], [1160, 543], [1158, 601], [1120, 593]]}, {"transcription": "LESS?", "points": [[1164, 542], [1252, 545], [1253, 624], [1166, 602]]}, {"transcription": "Singapore's", "points": [[1032, 177], [1185, 73], [1191, 143], [1038, 223]]}, {"transcription": "no.1", "points": [[1190, 73], [1270, 19], [1278, 91], [1194, 133]]}]
```

**Reply by liuhongen1234567** [reply]
For end-to-end evaluation, you can refer to E2EMetric in PGNet https://github.com/PaddlePaddle/PaddleOCR/blob/fd5b4e1049b758cf29b3c922a19b4c5f4ec47b88/configs/e2e/e2e_r50_vd_pg.yml#L63

**Reply by Jeep-Z** [reply]
For end-to-end evaluation, I want to evaluate detection and recognition in pipeline. Thx, I will evaluate det & rec one by one.

**Reply by liuhongen1234567** [reply]
Hello, PaddleOCR provides complete end-to-end inference code, such as python tools/infer/predict_system.py --image_dir="testB" --det_model_dir="./ch_PP-OCRv4_det_infer/" --rec_model_dir="./ch_PP-OCRv4_rec_infer/" --draw_img_save_dir="output" or the PP-OCRv5 pipeline https://paddlepaddle.github.io/PaddleOCR/main/en/version3.x/pipeline_usage/OCR.html#1-ocr-pipeline-introduction. Unfortunately, as far as I know, there is no corresponding end-to-end evaluation code provided. You can refer to E2EMetric and the end-to-end output results to implement the end-to-end evaluation yourself.
